### PR TITLE
encryption: cleanup and fix static pod deployer nodeName logic

### PR DESCRIPTION
**Pull request by sttts**
We compared the pod names, not the node name they were scheduled to.

### CODE REVIEW

File: pkg/operator/encryption/deployer/staticpod.go [hunk 1]
```diff
@@ -194,6 +177,46 @@ func getAPIServerRevisionOfAllInstances(nodes []operatorv1.NodeStatus, podClient
...
+	for _, p := range pods {
+		ret.Insert(p.Spec.NodeName)
+	}
+	return ret
+}
+
+func categorizePods(pods []corev1.Pod) (good []*corev1.Pod, bad []*corev1.Pod, progressing bool, err error) {
+	for _, apiServerPod := range pods {
+		switch phase := apiServerPod.Status.Phase; phase {
+		case corev1.PodRunning: // TODO check that total running == number of masters?
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/staticpod.go]: rm `TODO` as this is what this PR does.

File: pkg/operator/encryption/deployer/staticpod.go [hunk 2]
```diff
@@ -123,65 +120,51 @@ const revisionLabel = "revision"
...
-			return "", fmt.Errorf("api server pod %s in unknown phase", apiServerPod.Name)
-		case corev1.PodSucceeded, corev1.PodFailed:
-			// handle failed pods carefully to make sure things are healthy
-			// since the API server should never exit, a succeeded pod is considered as failed
-			failed.Insert(apiServerPod.Labels[revisionLabel])
-		default:
-			// error in case new unexpected phases get added
-			return "", fmt.Errorf("api server pod %s has unexpected phase %v", apiServerPod.Name, phase)
-		}
+	good, bad, progressing, err := categorizePods(apiServerPods.Items)
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/staticpod.go]: I like the names

File: pkg/operator/encryption/deployer/staticpod.go [hunk 3]
```diff
@@ -123,65 +120,51 @@ const revisionLabel = "revision"
 // Once a converged revision has been determined, it can be used to determine
 // what encryption config state has been successfully observed by the API servers.
 // It assumes that podClient is doing live lookups against the cluster state.
-func getAPIServerRevisionOfAllInstances(nodes []operatorv1.NodeStatus, podClient corev1client.PodInterface) (string, error) {
+func getAPIServerRevisionOfAllInstances(revisionLabel string, nodes []operatorv1.NodeStatus, podClient corev1client.PodInterface) (string, error) {
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/staticpod.go]: nit: this arg `revisionLabel`could be internal to this function `getAPIServerRevisionOfAllInstances`, no need to expose it.

File: pkg/operator/encryption/deployer/staticpod.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ func (d *StaticPodDeployer) DeployedEncryptionConfigSecret() (secret *corev1.Sec
 		return nil, false, nil
 	}
 
-	revision, err := getAPIServerRevisionOfAllInstances(status.NodeStatuses, d.podClient)
+	revision, err := getAPIServerRevisionOfAllInstances("revision", status.NodeStatuses, d.podClient)
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/staticpod.go]: does `NodeStatuses` contain only master nodes?

File: pkg/operator/encryption/deployer/staticpod.go (hunk 3 above)
Comment (sttts) [reply pkg/operator/encryption/deployer/staticpod.go]: it's intentional. A static-pod deployer will use  a different label.

File: pkg/operator/encryption/deployer/staticpod.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/encryption/deployer/staticpod.go]: oh, didn't notice we had that TODO

File: pkg/operator/encryption/deployer/staticpod.go (hunk 4 above)
Comment (sttts) [reply pkg/operator/encryption/deployer/staticpod.go]: yes

**Pull request merged**